packages/noya-state/src/reducers/themeReducer.ts

```diff
@@ -385,6 +394,33 @@ export function themeReducer(
         });
       });
     }
+    case 'setSymbolName': {
+      const [, id, name] = action;
+
+      const ids = typeof id === 'string' ? [id] : id;
+
+      return produce(state, (draft) => {
+        const symbols: Sketch.SymbolMaster[] = [];
+
+        draft.sketch.pages.forEach((p) =>
```

Comment:
Can we call the `getSymbols` selector here? E.g. `getSymbols(state).filter(symbol => ids.includes(symbol.do_objectID)`

Reply:
does it work? don't we need tou use the draft that has the type `WritableDraft<ApplicationState>` and it can't be use in that function 🤔

Reply:
I wasn't sure, so I tried it and it seems OK:

![Screen Shot 2021-05-18 at 1 13 59 PM](https://user-images.githubusercontent.com/1198882/118717433-eb197600-b7da-11eb-9847-b1434dc5595f.png)

I'm actually not sure when the `WritableDraft` stuff is necessary. I think it's a good idea to use that type on any functions that _mutate_ the state, but shouldn't matter for functions that don't.